plugins/modules/user_role_workflow_manager.py

```diff
@@ -1277,19 +1338,22 @@ def get_diff_merged(self, config):
 
                 if not consolidated_data:
                     self.msg = "User with username '{0}' already exists and does not require an update.".format(self.have.get("username"))
+                    self.no_update_user.append(self.have.get("username"))
                     self.log(self.msg, "INFO")
                     responses["role_operation"] = {"response": config}
                     self.result["response"] = self.msg
                     self.status = "success"
                     return self
 
                 if update_required_param.get("role_list"):
-                    user_in_have = self.have["current_user_config"]
-                    update_param = update_required_param
-                    update_param["username"] = user_in_have.get("username")
-                    update_param["user_id"] = user_in_have.get("user_id")
-                    user_info_params = self.snake_to_camel_case(update_param)
-                    task_response = self.update_user(user_info_params)
+                    if self.want["username"] not in self.have["current_user_config"]["username"]:
+                        task_response = {"error_message": "Username for an existing User cannot be updated."}
```

Comment:
task_response = {"error_message": "Username for an existing user cannot be updated."}

Reply:
Updated the code accordingly.